### ISSUE TITLE
Add ObjectiveC common types

### ISSUE THREAD

**Issue by dcharkes** [dart-lang/ffi#136]
Context: We'd like to add ObjectiveC bindings to `package:ffigen`:

* https://github.com/dart-lang/ffigen/issues/279

To make `NativeTypes` for Objective C bindings reusable when multiple bindings are generated, they should live in a package instead of being generated.

For example for pointers to objects we could generate `Pointer<NSObject>` instead of `Pointer<Opaque>`.

This package is a good place, because it is FFI helpers.

@mit-mit any thoughts. Should we just put it in here?

cc @liamappelbe

**Comment by mit-mit** [comment]
Perhaps here, but in a seperate library (`import 'package:ffi/objectivec.dart'`)? @lrhn wdyt?

**Comment by dcharkes** [comment]
> They won't cost anyone anything unless they use the types. (Right?)

I believe that is correct.

@liamappelbe Any need for names that do not have `ObjectiveC` in the name?

**Comment by mit-mit** [comment]
> If not, I'm fine with putting them in the main library and documenting them as "Objective C/Swift only". They won't cost anyone anything unless they use the types. (Right?)

Won't they still show up in code completions? That is the main way many developers learn an API, and it would be confusing if ObjC completions show up for someone doing regular C interop...

**Comment by liamappelbe** [comment]
> Any need for names that do not have ObjectiveC in the name?

No, that's fine, we could prefix all the new types with ObjectiveC or ObjC

**Comment by liamappelbe** [comment]
Objective C's type names are a bit of a mess. I want to add dart equivalents of  `id`, `SEL`, `NSObject`, `Class`, and `Block`. So some of these definitely have potential to clash with other types if we name them exactly what they're called in ObjC, but the naming is so inconsistent that I think it makes sense to just give them more sensible names.

So I'll put them in the main library, but call them `ObjCObject`, `ObjCSel`, `ObjCClass`, and `ObjCBlock`.
